### PR TITLE
8339992: RISC-V: some minor improvements of base64_vector_decode_round

### PR DESCRIPTION
**Pull request by Hamlin-Li**
Hi,
Can you help to review this simple improvement?
Thanks.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339992](https://bugs.openjdk.org/browse/JDK-8339992): RISC-V: some minor improvements of base64_vector_decode_round (**Enhancement** - P4)

### Reviewers
 * [Ludovic Henry](https://openjdk.org/census#luhenry) (@luhenry - Committer) Review applies to [9fa797a7](https://git.openjdk.org/jdk/pull/21059/files/9fa797a7c12b0be1f19b3d1c919cd1cdb73e0612)
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21059/head:pull/21059` \
`$ git checkout pull/21059`

Update a local copy of the PR: \
`$ git checkout pull/21059` \
`$ git pull https://git.openjdk.org/jdk.git pull/21059/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21059`

View PR using the GUI difftool: \
`$ git pr show -t 21059`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21059.diff">https://git.openjdk.org/jdk/pull/21059.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21059#issuecomment-2358414258)

### CODE REVIEW

**Comment by Hamlin-Li** [comment]
@RealFYang As we discussed previously, we're supposed to do following change too,
* `vsetvli(x0, failedIdx, Assembler::e8, lmul, Assembler::mu, Assembler::tu);` => remove mu/tu (i.e. change to ma/ta)

But after study the spec again, I can not find out the exact word supporting this change (maybe I missed it), so for safety consideration I will leave it as is.

If someone can find out the exact evidence in the spec that changing from mu/tu => mu/tu is absolutely safe on all devices, please kindly share your information, I'll add the change.

**Review by luhenry — APPROVED** [review]

**Review by RealFYang — APPROVED** [review]
Looks good. Thanks for making this change.

File: src/hotspot/cpu/riscv/stubGenerator_riscv.cpp [hunk 1]
```diff
@@ -5358,8 +5358,11 @@ class StubGenerator: public StubCodeGenerator {
     __ vor_vv(outputV1, outputV1, outputV2);
     __ vmseq_vi(v0, outputV1, -1);
     __ vfirst_m(failedIdx, v0);
-    Label NoFailure;
-    __ beq(failedIdx, minusOne, NoFailure);
+    Label NoFailure, FailureAt0Idx;
```
Comment (RealFYang) [inline src/hotspot/cpu/riscv/stubGenerator_riscv.cpp:5361]: Nit: Maybe rename `FailureAt0Idx` to `FailureAtIdx0`?

Comment (Hamlin-Li) [reply src/hotspot/cpu/riscv/stubGenerator_riscv.cpp]: Yeh, it makes sense!

**Review by RealFYang — APPROVED** [review]

**Comment by Hamlin-Li** [comment]
Thanks for your reviewing! @luhenry @RealFYang 

/integrate

**Pull request closed without merging**